### ISSUE TITLE
Deadlock when starting behind authenticated proxy

### ISSUE THREAD

**Issue by rscadrde** [oracle/visualvm#48]
Start visualvm 1.4 behind an authenticated proxy and a few seconds later application freezes.

Workaround: Quickly open "tools-settings" and switch off proxy.

**Comment by rscadrde** [comment]
EventQueue hangs here:
```
Thread [AWT-EventQueue-0] (Suspended)
waiting for: org.netbeans.ModuleManager$SystemClassLoader (id=6734)
org.netbeans.ModuleManager$SystemClassLoader.getResourcesImpl(java.lang.String) line: 700
org.netbeans.ModuleManager$SystemClassLoader(org.netbeans.ProxyClassLoader).getResources(java.lang.String) line: 390
org.openide.util.lookup.MetaInfServicesLookup.search(java.lang.Class<?>, java.util.Collection<org.openide.util.lookup.AbstractLookup.Pair<?>>) line: 199
org.openide.util.lookup.MetaInfServicesLookup.beforeLookup(org.openide.util.Lookup.Template<?>) line: 156
org.openide.util.lookup.MetaInfServicesLookup.beforeLookupResult(org.openide.util.Lookup.Template<?>) line: 135
org.openide.util.lookup.MetaInfServicesLookup(org.openide.util.lookup.AbstractLookup).lookup(org.openide.util.Lookup.Template<T>) line: 483
org.openide.util.lookup.ProxyLookup$R.initResults() line: 390
org.openide.util.lookup.ProxyLookup$R.myBeforeLookup(boolean, boolean) line: 669
org.openide.util.lookup.ProxyLookup$R.computeResult(int, boolean) line: 549
org.openide.util.lookup.ProxyLookup$R.allInstances(boolean) line: 509
org.openide.util.lookup.ProxyLookup$R.allInstances() line: 505
org.netbeans.core.startup.MainLookup(org.openide.util.Lookup).lookupAll(java.lang.Class<T>) line: 271
org.netbeans.core.windows.services.DialogDisplayerImpl.customizeDlg(org.netbeans.core.windows.services.NbPresenter) line: 332
org.netbeans.core.windows.services.DialogDisplayerImpl.access$000(org.netbeans.core.windows.services.NbPresenter) line: 76
org.netbeans.core.windows.services.DialogDisplayerImpl$1AWTQuery.showDialog() line: 262
org.netbeans.core.windows.services.DialogDisplayerImpl$1AWTQuery.run() line: 192
java.awt.event.InvocationEvent.dispatch() line: 311
java.awt.EventQueue.dispatchEventImpl(java.awt.AWTEvent, java.lang.Object) line: 756
java.awt.EventQueue.access$500(java.awt.EventQueue, java.awt.AWTEvent, java.lang.Object) line: 97
java.awt.EventQueue$3.run() line: 709
java.awt.EventQueue$3.run() line: 703
java.security.AccessController.doPrivileged(java.security.PrivilegedAction<T>, java.security.AccessControlContext) line: not available [native method]
java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(java.security.PrivilegedAction<T>, java.security.AccessControlContext, java.security.AccessControlContext) line: 80
java.awt.EventQueue.dispatchEvent(java.awt.AWTEvent) line: 726
java.awt.EventDispatchThread.pumpOneEventForFilters(int) line: 201
java.awt.EventDispatchThread.pumpEventsForFilter(int, java.awt.Conditional, java.awt.EventFilter) line: 116
java.awt.EventDispatchThread.pumpEventsForHierarchy(int, java.awt.Conditional, java.awt.Component) line: 105
java.awt.EventDispatchThread.pumpEvents(int, java.awt.Conditional) line: 101
java.awt.EventDispatchThread.pumpEvents(java.awt.Conditional) line: 93
java.awt.EventDispatchThread.run() line: 82

```

Corresponding Thread that holds the Lock: 

```
System Thread [pool-4-thread-1] (Suspended)	
owns: org.netbeans.core.NbAuthenticator (id=6738)	
owns: java.lang.Class (sun.net.www.protocol.http.NegotiateAuthentication) (id=3137)	
owns: org.netbeans.ModuleManager$SystemClassLoader (id=6734)	
owns: sun.net.www.protocol.https.DelegateHttpsURLConnection (id=6739)	
waiting for: org.netbeans.core.windows.services.DialogDisplayerImpl$1AWTQuery (id=6737)	
java.lang.Object.wait(long) line: not available [native method]	
org.netbeans.core.windows.services.DialogDisplayerImpl.notify(org.openide.NotifyDescriptor, boolean) line: 297	
org.netbeans.core.windows.services.DialogDisplayerImpl.notify(org.openide.NotifyDescriptor) line: 173	
org.netbeans.core.NbAuthenticator.getPasswordAuthentication() line: 101	
java.net.Authenticator.requestPasswordAuthentication(java.lang.String, java.net.InetAddress, int, java.lang.String, java.lang.String, java.lang.String, java.net.URL, java.net.Authenticator$RequestorType) line: 317	
sun.net.www.protocol.http.spnego.NegotiateCallbackHandler.getAnswer() line: 65	
sun.net.www.protocol.http.spnego.NegotiateCallbackHandler.handle(javax.security.auth.callback.Callback[]) line: 86	
com.sun.security.auth.module.Krb5LoginModule.promptForName(boolean) line: 858	
com.sun.security.auth.module.Krb5LoginModule.attemptAuthentication(boolean) line: 704	
com.sun.security.auth.module.Krb5LoginModule.login() line: 617	
sun.reflect.NativeMethodAccessorImpl.invoke0(java.lang.reflect.Method, java.lang.Object, java.lang.Object[]) line: not available [native method]	
sun.reflect.NativeMethodAccessorImpl.invoke(java.lang.Object, java.lang.Object[]) line: 62	
sun.reflect.DelegatingMethodAccessorImpl.invoke(java.lang.Object, java.lang.Object[]) line: 43	
java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object...) line: 498	
javax.security.auth.login.LoginContext.invoke(java.lang.String) line: 755	
javax.security.auth.login.LoginContext.access$000(javax.security.auth.login.LoginContext, java.lang.String) line: 195	
javax.security.auth.login.LoginContext$4.run() line: 682	
javax.security.auth.login.LoginContext$4.run() line: 680	
java.security.AccessController.doPrivileged(java.security.PrivilegedExceptionAction<T>, java.security.AccessControlContext) line: not available [native method]	
javax.security.auth.login.LoginContext.invokePriv(java.lang.String) line: 680	
javax.security.auth.login.LoginContext.login() line: 587	
sun.security.jgss.GSSUtil.login(sun.security.jgss.GSSCaller, org.ietf.jgss.Oid) line: 258	
sun.security.jgss.krb5.Krb5Util.getTicket(sun.security.jgss.GSSCaller, java.lang.String, java.lang.String, java.security.AccessControlContext) line: 158	
sun.security.jgss.krb5.Krb5InitCredential$1.run() line: 335	
sun.security.jgss.krb5.Krb5InitCredential$1.run() line: 331	
java.security.AccessController.doPrivileged(java.security.PrivilegedExceptionAction<T>) line: not available [native method]	
sun.security.jgss.krb5.Krb5InitCredential.getTgt(sun.security.jgss.GSSCaller, sun.security.jgss.krb5.Krb5NameElement, int) line: 330	
sun.security.jgss.krb5.Krb5InitCredential.getInstance(sun.security.jgss.GSSCaller, sun.security.jgss.krb5.Krb5NameElement, int) line: 145	
sun.security.jgss.krb5.Krb5MechFactory.getCredentialElement(sun.security.jgss.spi.GSSNameSpi, int, int, int) line: 122	
sun.security.jgss.krb5.Krb5MechFactory.getMechanismContext(sun.security.jgss.spi.GSSNameSpi, sun.security.jgss.spi.GSSCredentialSpi, int) line: 187	
sun.security.jgss.GSSManagerImpl.getMechanismContext(sun.security.jgss.spi.GSSNameSpi, sun.security.jgss.spi.GSSCredentialSpi, int, org.ietf.jgss.Oid) line: 224	
sun.security.jgss.GSSContextImpl.initSecContext(java.io.InputStream, java.io.OutputStream) line: 212	
sun.security.jgss.GSSContextImpl.initSecContext(byte[], int, int) line: 179	
sun.security.jgss.spnego.SpNegoContext.GSS_initSecContext(byte[]) line: 882	
sun.security.jgss.spnego.SpNegoContext.initSecContext(java.io.InputStream, int) line: 317	
sun.security.jgss.GSSContextImpl.initSecCon
 sun.security.jgss.GSSContextImpl.initSecContext(java.io.InputStream, java.io.OutputStream) line: 212    
sun.security.jgss.GSSContextImpl.initSecContext(byte[], int, int) line: 179    
sun.security.jgss.spnego.SpNegoContext.GSS_initSecContext(byte[]) line: 882    
sun.security.jgss.spnego.SpNegoContext.initSecContext(java.io.InputStream, int) line: 317    
sun.security.jgss.GSSContextImpl.initSecContext(java.io.InputStream, java.io.OutputStream) line: 248    
sun.security.jgss.GSSContextImpl.initSecContext(byte[], int, int) line: 179    
sun.net.www.protocol.http.spnego.NegotiatorImpl.init(sun.net.www.protocol.http.HttpCallerInfo) line: 108    
sun.net.www.protocol.http.spnego.NegotiatorImpl.<init>(sun.net.www.protocol.http.HttpCallerInfo) line: 117    
sun.reflect.NativeConstructorAccessorImpl.newInstance0(java.lang.reflect.Constructor<?>, java.lang.Object[]) line: not available [native method]    
sun.reflect.NativeConstructorAccessorImpl.newInstance(java.lang.Object[]) line: 62    
sun.reflect.DelegatingConstructorAccessorImpl.newInstance(java.lang.Object[]) line: 45    
java.lang.reflect.Constructor.newInstance(java.lang.Object...) line: 423    
sun.net.www.protocol.http.Negotiator.getNegotiator(sun.net.www.protocol.http.HttpCallerInfo) line: 63    
sun.net.www.protocol.http.NegotiateAuthentication.isSupportedImpl(sun.net.www.protocol.http.HttpCallerInfo) line: 138    
sun.net.www.protocol.http.NegotiateAuthentication.isSupported(sun.net.www.protocol.http.HttpCallerInfo) line: 111    
sun.net.www.protocol.http.AuthenticationHeader.parse(java.util.Set<java.lang.String>) line: 200    
sun.net.www.protocol.http.AuthenticationHeader.<init>(java.lang.String, sun.net.www.MessageHeader, sun.net.www.protocol.http.HttpCallerInfo, boolean, java.util.Set<java.lang.String>) line: 143    
sun.net.www.protocol.https.DelegateHttpsURLConnection(sun.net.www.protocol.http.HttpURLConnection).doTunneling() line: 2097    
sun.net.www.protocol.https.DelegateHttpsURLConnection(sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection).connect() line: 183    
sun.net.www.protocol.https.HttpsURLConnectionImpl.connect() line: 162    
org.netbeans.modules.autoupdate.updateprovider.NetworkAccess$Task$2.call() line: 166    
org.netbeans.modules.autoupdate.updateprovider.NetworkAccess$Task$2.call() line: 148    
java.util.concurrent.FutureTask.run() line: 266    
java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker) line: 1149    
java.util.concurrent.ThreadPoolExecutor$Worker.run() line: 624    
java.lang.Thread.run() line: 748 
```

**Comment by thurka** [comment]
This looks similar to NetBeans [bug #248308](https://netbeans.org/bugzilla/show_bug.cgi?id=248308) and it looks like the root problem is in JDK - reported as [JDK-8068184](https://bugs.openjdk.java.net/browse/JDK-8068184). Additional info seems to be available also [here](https://issues.apache.org/jira/browse/NETBEANS-58)

**Comment by thurka** [comment]
The [JDK-8068184](https://bugs.openjdk.java.net/browse/JDK-8068184) is fixed in JDK 11.0.7 and JDK 8u252.
